hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/JoinHashPhysicalRel.java

```diff
@@ -58,9 +58,34 @@ public Join copy(
         return new JoinHashPhysicalRel(getCluster(), traitSet, left, right, conditionExpr, joinType);
     }
 
+    /**
+     * Cost calculation of Hash Join relation. It does not rely on children cost.
+     * <p>
+     * Hash Join algorithm is a more advanced join algorithm, which builds a hash table for the left
+     * row set, and then compare each row from the right side. Cost estimation is the following: <ol>
+     * <li> Produced row count is L * R assuming the join selectivity is 1.
+     * <li> Processed row count is L + R because we traverse both sides once per JOIN.
+     * <li> CPU is L * (hash table build cost) + R * (row comparison cost). </ol>
+     * <p>
+     * A perfect estimation must also include memory (occupied by the hash table) and IO costs,
+     * as well as a selectivity for the right row set.
```

Comment:
```suggestion
     * <li> Produced row count is L * R * <i>join selectivity</i>.
     * <li> Processed row count is L + R because we traverse both sides once per join.
     * <li> CPU is L * <i>hash table build cost</i> + R * <i>row comparison cost</i>. </ol>
     * <p>
     * A perfect estimation must also include memory (occupied by the hash table) and IO costs.
```

Reply:
Same as below.